Core/Src/main.c

```diff
@@ -87,17 +91,37 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);  // Start PWM on channel 1
+  uint8_t rxData[3];
+  uint8_t txData[3];
```

Comment:
Use the benefit of array- You could say const uint8_t txData[3]= {0x1, 0x80, 0} instead of adding lines for each element in the array.

Reply:
yeah that makes sense made that change